src/hooks/worlds/useWorldsByNotHidden.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from "react";
+import { collection, getFirestore, query, where } from "firebase/firestore";
+
+import { ALWAYS_EMPTY_ARRAY, COLLECTION_WORLDS, FIELD_HIDDEN } from "settings";
+
+import { World } from "api/world";
+
+import { LoadStatus } from "types/fire";
+import { WorldId, WorldWithId } from "types/id";
+
+import { withIdConverter } from "utils/converters";
+
+import { useLiveQuery } from "hooks/fire/useLiveQuery";
+
+type UseWorldsByNotHidden = () => LoadStatus & { worlds: WorldWithId[] };
+
+export const useWorldsByNotHidden: UseWorldsByNotHidden = () => {
+  const memoizedQuery = useMemo(
+    () =>
+      query(
+        collection(getFirestore(), COLLECTION_WORLDS),
+        where(FIELD_HIDDEN, "==", false)
+      ).withConverter(withIdConverter<World, WorldId>()),
+    []
+  );
+
+  const { error, isLoading, isLoaded, data } = useLiveQuery(memoizedQuery);
+
+  return useMemo(
+    () => ({ error, isLoading, isLoaded, worlds: data ?? ALWAYS_EMPTY_ARRAY }),
+    [error, isLoading, isLoaded, data]
+  );
+};
```

Comment:
I don't think this should exist at all. Hidden worlds should be invisible from **all** queries.

Reply:
So far `  .where("isHidden", "==", false)` can be found in half of dozen lines. It is one that already exists and  `useWorldsByNotHidden` just copied the query:

```ts
export const useOwnWorlds: UseOwnWorlds = ({ userId }) => {
  const firestore = useFirestore();

  const ownWorldsRef = query(
    collection(firestore, COLLECTION_WORLDS),
    where("isHidden", "==", false),
    where("owners", "array-contains", userId)
  ).withConverter(withIdConverter<WorldWithId, WorldId>());
```

Is there a way of reading worlds from DB without putting `isHidden` constraint but not having hidden worlds in the result?

Reply:
No :(